lib/view_component/compiler.rb

```diff
@@ -31,6 +31,8 @@ def compile(raise_errors: false, force: false)
       return if compiled? && !force
       return if component_class == ViewComponent::Base
 
+      component_class.superclass.compile(raise_errors: raise_errors) if should_compile_superclass?
```

Comment:
I could be wrong, but I believe the parent could already have been recompiled (e.g. the parent itself or another component that inherits from it has been rendered since its sidecar file changed), in which case another recompilation would be unnecessary.

Reply:
@camertron Am I incorrect?

Reply:
Ah ok that makes sense, I get it 👍